pkg/ccl/changefeedccl/encoder.go

```diff
@@ -199,8 +199,11 @@ func (e *jsonEncoder) EncodeValue(_ context.Context, row encodeRow) ([]byte, err
 	var after map[string]interface{}
 	if !row.deleted {
 		columns := row.tableDesc.PublicColumns()
```

Comment:
I considered adding a method to the TableDescriptor that only stores non-virtual columns, but I believe that this would require us to extend our column cache to store non-virtual columns. I am not sure on how expensive this would be, but I would love to get some thoughts on whether or not we should go in that direction (as apposed to filtering out virtual columns that are returned from `PublicColumns()`).